demos/common/cpp/pipelines/src/async_pipeline.cpp

```diff
@@ -14,26 +14,29 @@
 // limitations under the License.
 */
 
-#include "pipelines/async_pipeline.h"
+#include <memory>
+#include <string>
+#include <openvino/openvino.hpp>
 #include <utils/common.hpp>
 #include <utils/slog.hpp>
+#include "pipelines/async_pipeline.h"
 
-AsyncPipeline::AsyncPipeline(std::unique_ptr<ModelBase>&& modelInstance, const CnnConfig& cnnConfig, InferenceEngine::Core& core) :
+AsyncPipeline::AsyncPipeline(std::unique_ptr<ModelBase>&& modelInstance, const ModelConfig& config, ov::runtime::Core& core) :
     model(std::move(modelInstance)) {
-    execNetwork = model->loadExecutableNetwork(cnnConfig, core);
+    compiledModel = model->compileModel(config, core);
     // --------------------------- Create infer requests ------------------------------------------------
-    unsigned int nireq = cnnConfig.maxAsyncRequests;
+    unsigned int nireq = config.maxAsyncRequests;
     if (nireq == 0) {
         try {
             // +1 to use it as a buffer of the pipeline
-            nireq = execNetwork.GetMetric(METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS)).as<unsigned int>() + 1;
-        } catch (const InferenceEngine::Exception& ex) {
-            throw std::runtime_error(std::string("Every device used with the demo should support "
-                "OPTIMAL_NUMBER_OF_INFER_REQUESTS ExecutableNetwork metric. Failed to query the metric with error: ") + ex.what());
+            nireq = compiledModel.get_property(METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS)).as<unsigned int>() + 1;
+        } catch (const ov::Exception& ex) {
+            throw std::runtime_error(std::string("Every device used with the demo should support compiled model's property "
+                "\"OPTIMAL_NUMBER_OF_INFER_REQUESTS\". Failed to query the property with error: ") + ex.what());
```

Comment:
```suggestion
                "'OPTIMAL_NUMBER_OF_INFER_REQUESTS'. Failed to query the property with error: ") + ex.what());
```
With that you don't need `\`

Reply:
why so afraid of escape sequences ?

Reply:
Removed.